samples/word/50-document/manage-custom-xml-part.yaml

```diff
@@ -0,0 +1,222 @@
+              customXmlPart.insertElement(
+                "/Reviewers",
+                "<Lead>Mark</Lead>",
+                { contoso: "http://schemas.contoso.com/review/1.0" },
+                0
+              );
+              const xmlBlob = customXmlPart.getXml();
+              await context.sync();
+
+              const readableXml = addLineBreaksToXML(xmlBlob.value);
+              console.log("Successfully inserted element:");
+              console.log(readableXml);
+            } else {
+              console.warn("No custom XML part to insert element into");
+            }
+          });
+        }
+
+        async function deleteCustomXmlPart() {
+          // Delete the specified custom XML part.
+          await Word.run(async (context) => {
+            const settings = context.document.settings;
+            const xmlPartIDSetting = settings.getItemOrNullObject("ContosoReviewXmlPartId").load("value");
+            await context.sync();
+
+            if (xmlPartIDSetting.value) {
+              let customXmlPart = context.document.customXmlParts.getItem(xmlPartIDSetting.value);
+              const xmlBlob = customXmlPart.getXml();
+              customXmlPart.delete();
+              customXmlPart = context.document.customXmlParts.getItemOrNullObject(xmlPartIDSetting.value);
+
+              await context.sync();
+
+              if (customXmlPart.isNullObject) {
+                console.log(`The XML part with the id ${xmlPartIDSetting.value} has been deleted`);
+
+                // Delete the unneeded setting too.
+                xmlPartIDSetting.delete();
+
+                await context.sync();
+              } else {
+                const readableXml = addLineBreaksToXML(xmlBlob.value);
+                const strangeMessage = `This is strange. The XML part with the id ${xmlPartIDSetting.value} wasn't deleted:\n${readableXml}`;
+                console.error(strangeMessage);
+              }
+            } else {
+              console.warn("No custom XML part to delete");
+            }
+          });
+        }
+
+        function addLineBreaksToXML(xmlBlob: string): string {
+          const replaceValue = new RegExp(">");
+          return xmlBlob.replace(/></g, "> <");
+        }
+
+        /** Default helper for invoking an action and handling errors. */
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            <p>This sample shows how to add, query, edit, and delete a custom XML part in a document.</p>
+        </section>
+
+        <section class="samples ms-font-m">
+            <h3>Try it out</h3>
+            <p>Add a custom XML part</p>
```

Comment:
I'm not sure this text adds anything. Consider removing these lines and letting the buttons stand alone.